dir.props

```diff
@@ -16,6 +16,7 @@
     <PackageOutputRoot Condition="'$(PackageOutputRoot)'==''">$(BinDir)packages/</PackageOutputRoot>
 
     <!-- Input Directories -->
+    <PackagesDir>$(RestorePackagesPath)</PackagesDir>
```

Comment:
I think `RestorePackagesPath` should have the `DotNet` prefix.

Reply:
I can be persuaded here, but my thinking/reasoning was that when a property exists in the product that matches exactly what the intention of a property we are using, we should use the official property.  This way, in the simple cases, the repo doesn't need to do anything special to respect the property.

For example, when passing `$(Configuration)` to the submodules, I don't think it should be `$(DotNetConfiguration)`.

[`$(RestorePackagesPath)`](https://docs.microsoft.com/en-us/nuget/schema/msbuild-targets) is the official property for where NuGet should restore packages to.

The reason `$(DotNetSourceBuildRestoreSources)` isn't `$(RestoreSources)` is because we want to allow the repo to add more sources, which MSBuild doesn't allow when specifying a property on the command line.

Thoughts?

Reply:
I can also see it going either way. This line in the (current) repo api framework doc makes it seem more black and white:

> In order to avoid conflicts between existing MSBuild properties used by the repos and the properties passed in by source-build, source-build will append a prefix to every property it passes in: `DotNet`.

I didn't realize it was a NuGet msbuild property, that does make it seem more reasonable. It makes me a bit uneasy to say that repos need to respect properties for "assumed" toolsets in general, though.

Reply:
I kind of lean towards always prefixing to with DotNet here as well. Both for consistency in naming as well as the same reason you call out for the sources we looks any ability to mutate that property when we pass it at the command line.

Reply:
Just so I'm 100% clear on the consistency aspect - We wouldn't pass in `$(Configuration)` to the sub-repos, right?  Same for `$(Platform)`.  We would prefix these properties as well.

My real hesitation here is that doing this forces each repo to specifically handle these separate properties, when a good percentage of the repos wouldn't have to do anything special at all, if we pass in the official property used by the tooling.

Reply:
That is a fair point but couldn't we solve that by having a boiler-plate dir.props (or Directory.Build.props) that repos can include along with the boiler-plate build scripts?

Reply:
Why have something duplicated 24 times (and growing), when you don't have to?  

If we are going to do `$(Configuration)` this way, then I'm fine with the consistency argument.  But if we aren't, then the consistency argument doesn't hold weight IMO.

For the "ability to mutate that property" argument - I would thought we don't want repos to mutate this property.

Reply:
We haven't really gotten to the Configuration/Platform properties yet but we likely need to rename those as well. I know building corefx will be a challenge if those come in as global properties because we have to redefine Configuration as well. 

>For the "ability to mutate that property" argument - I would thought we don't want repos to mutate this property.

I think there are always cases where that is needed. For example corefx doesn't use the standard Configuration property because we need to encode other things like the OS and TFM into that property in order to support all the cross-compiling we need.

Reply:
Ok, then I'm fine with this property getting the prefix.

Reply:
We can have a (independent) BuildTools package to deliver a mapping props file. There's also the idea that's been floating around for a while of having the boilerplate scripts auto-updated, so maintenance would be mostly automatic anyway.